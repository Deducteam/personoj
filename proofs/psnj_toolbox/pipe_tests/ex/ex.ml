let () =
  let check = false in
  let out_files = [ "exi.lp" ] in
  let src = "ex.log" in
  Testing.run_pipe ~src ~out_files ~check Sys.argv.(1)
