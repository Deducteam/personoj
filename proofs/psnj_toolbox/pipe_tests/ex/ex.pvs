ex: THEORY
BEGIN
	P: bool
	exi: LEMMA EXISTS (x: bool): P IMPLIES P
END ex
