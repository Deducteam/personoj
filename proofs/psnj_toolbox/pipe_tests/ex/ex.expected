=> exi.lp:
require open pipe.encoding.lhol pipe.encoding.propositional_connectives;
require pipe.spec;
symbol exi!1 : @Prf (@imp (imp pipe.spec.P pipe.spec.P) (@imp (disj (not pipe.spec.P) pipe.spec.P) (@imp (imp pipe.spec.P pipe.spec.P) (ex prop (λ (_ : El prop), imp pipe.spec.P pipe.spec.P))))) ≔
begin
  why3;
end;
symbol exi!2 : @Prf (imp pipe.spec.P pipe.spec.P) ≔
begin
  why3;
end;
symbol exi!3 : @Prf (@imp (imp pipe.spec.P pipe.spec.P) (@imp (disj (not pipe.spec.P) pipe.spec.P) (@imp (imp pipe.spec.P pipe.spec.P) (imp pipe.spec.P pipe.spec.P)))) ≔
begin
  why3;
end;
symbol exi!4 : @Prf (disj (not pipe.spec.P) pipe.spec.P) ≔
begin
  why3;
end;
symbol exi!5 : @Prf (imp pipe.spec.P pipe.spec.P) ≔
begin
  why3;
end;
